src/utils/crc32.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-bitwise, operator-assignment */
+
+export default str => {
+  const buffer = new Array(256).fill(0).map((item, index) => {
+    let c = index;
+    for (let j = 0; j < 8; j = j + 1) {
+      c = c & 1 ? 0xedb88320 ^ (c >>> 1) : c >>> 1;
+    }
+    return c;
+  });
+
+  let crc = 0;
+  const cleanString = unescape(encodeURIComponent(str));
+  crc = crc ^ -1;
+  cleanString
+    .split("")
+    .map(char => char.charCodeAt(0))
+    .forEach(char => {
+      const y = (crc ^ char) & 0xff;
+      crc = (crc >>> 8) ^ buffer[y];
+    });
+  crc = crc ^ -1;
+  return (crc >>> 0).toString(16);
+};
```

Comment:
Where did you get this code? Is there any reason you went with it instead of http://crc32.nichabi.com/javascript-function.php?

Reply:
I got this code from there.
I made a few changes because of lint errors.
Added `.toString(16)` because the function was returning a decimal number in JS, and every other crc32 example was returning a `hex`.
I compared the out put of many strings with available crc32 online examples

Reply:
I removed fill and added a loop as it was before.